numba/targets/arraymath.py

```diff
@@ -3717,8 +3717,34 @@ def np_kaiser_impl(M, beta):
 def np_cross(a, b):
     if isinstance(a, types.Array) and isinstance(b, types.Array):
 
```

Comment:
I was thinking of doing calculating the shape of `cp` here, and have different implementations in the inner scope, but:
1. I cannot slice with `Ellipsis`
2. I was not getting shapes with `np.broadcast` as expected.

Reply:
The `shape` isn't available in this scope. You'll find that `a` and `b` these are not Numpy `ndarry`s but rather Numba type objects. While they do have `ndim` and `dtype`, the `shape` isn't supported.